google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.java

```diff
@@ -186,10 +185,10 @@ public ApiFuture<AppendRowsResponse> append(JSONArray jsonArr, long offset)
       throws IOException, DescriptorValidationException {
     // Handle schema updates in a Thread-safe way by locking down the operation
     synchronized (this) {
-      // Update schema only work when connection pool is not enabled.
-      if (this.streamWriter.getConnectionOperationType() == Kind.CONNECTION_WORKER
-          && this.streamWriter.getUpdatedSchema() != null) {
-        refreshWriter(this.streamWriter.getUpdatedSchema());
+      TableSchema updatedSchemaAndTime = this.streamWriter.getUpdatedSchema();
+      // Create a new stream writer internally if a new updated schema is reported from backend.
+      if (updatedSchemaAndTime != null && !this.tableSchema.equals(updatedSchemaAndTime)) {
```

Comment:
We can directly use streamWriter.getUpdatedSchema() != null?

Reply:
Done